src/modelmanager.cpp

```diff
@@ -828,6 +839,41 @@ void ModelManager::watcher(std::future<void> exitSignal) {
     SPDLOG_LOGGER_INFO(modelmanager_logger, "Stopped model manager thread");
 }
 
+void ModelManager::cleanerRoutine(uint32_t resourcesCleanupIntervalSec, uint32_t sequenceCleanerIntervalMinutes, std::future<void> cleanerExitSignal) {
+    SPDLOG_LOGGER_INFO(modelmanager_logger, "Started cleaner thread");
+
+    uint32_t resourcesWaitTime = resourcesCleanupIntervalSec;
+    uint32_t sequenceWaitTime = sequenceCleanerIntervalMinutes * 60;
```

Comment:
For testability purposes I think we should have interface for this method set with intervals sets to lower time precision - eg. milliseconds. This way you can execute more tests faster.